crates/nu-cli/src/util.rs

```diff
@@ -46,9 +63,12 @@ fn gather_env_vars(vars: impl Iterator<Item = (String, String)>, engine_state: &
     let mut has_pwd = false;
 
     // Write all the env vars into a fake file
-    for (name, val) in vars {
+    for (name, mut val) in vars {
         if name == "PWD" {
             has_pwd = true;
+            if let Some(p) = init_cwd.to_str() {
```

Comment:
I think instead of relying on detection of PWD environment variable, we should always force-set it to `init_cwd`. We can remove the whole `has_pwd` -- related logic and move these three lines after the for loop (and report error if `to_str()` fails).

Reply:
Yeah, that's much better